docs/source/en/pipeline_tutorial.mdx

```diff
@@ -33,101 +33,202 @@ While each task has an associated [`pipeline`], it is simpler to use the general
+[`pipeline`] supports many parameters, some are task specific, some are general for all pipelines.
+In general you can specify parameters anywhere you want:
 
 ```py
->>> from transformers import pipeline
+generator(model="openai/whisper-large", my_parameter=1)
+out = generate(...)  # This will use `my_parameter=1`.
+out = generate(..., my_parameter=2)  # This will override and use `my_parameter=2`.
+out = generate(...)  # This will go back to using `my_parameter=1`.
+```
+
+Let's check out 3 important ones:
+
+### Device
 
->>> generator = pipeline(task="text-generation", model=model, tokenizer=tokenizer)
+If you use `device=n` the pipeline will automatically put the model on said device.
+This will work regardless wether you are using `PyTorch` or `Tensorflow`.
+
+```py
+generator(model="openai/whisper-large", device=0)
 ```
 
-Pass your input text to the [`pipeline`] to generate some text:
+If the model is too large for a single GPU, you can also use [`accelerate`](https://huggingface.co/docs/accelerate/index) (`device_map="auto"`).
 
 ```py
->>> generator(
-...     "Three Rings for the Elven-kings under the sky, Seven for the Dwarf-lords in their halls of stone"
-... )  # doctest: +SKIP
-[{'generated_text': 'Three Rings for the Elven-kings under the sky, Seven for the Dwarf-lords in their halls of stone, Seven for the Dragon-lords (for them to rule in a world ruled by their rulers, and all who live within the realm'}]
+#!pip install accelerate
+generator(model="openai/whisper-large", device_map="auto")
 ```
 
-## Audio pipeline
+### Batch size
 
-The [`pipeline`] also supports audio tasks like audio classification and automatic speech recognition.
+By default, pipelines will not batch inference for reasons explained in detail [here](https://huggingface.co/docs/transformers/main_classes/pipelines#pipeline-batching). The reason is that batching is not necessarily faster, and can actually be quite slower in some cases.
 
-For example, let's classify the emotion in this audio clip:
+But if it works in your use case, you can use:
 
 ```py
->>> from datasets import load_dataset
->>> import torch
+generator(model="openai/whisper-large", device=0, batch_size=2)
+audio_filenames = [f"audio_{i}.flac" for i in range(10)]
+texts = generator(audio_filenames)
+```
+
+This will actually run the pipeline on all 10 audio files you are providing, but it will pass them by batches of 2
+to the model (which is on GPU where batching is more likely to help), without any further code from you. 
+The output should always match what you would have received without batching. It is only meant as a way to help 
+get more speed out of the pipeline. 
+
+Pipelines alleviate some complexity of batching, because for some pipelines, 
+a single item (like a long audio file) need to be chunked into multiple parts
+for processing by a model. Pipeline [abstract that away for you](./main_classes/pipelines#pipeline-chunk-batching)
```

Comment:
```suggestion
Pipelines can also alleviate some of the complexities of batching because, for some pipelines, a single item (like a long audio file) needs to be chunked into multiple parts to be processed by a model. The pipeline performs this [*chunk batching*](./main_classes/pipelines#pipeline-chunk-batching) for you.
```